File: modules/agent/src/schemas/job.py
"""Job schemas for API validation."""

from typing import List, Optional

from pydantic import BaseModel


class JobCreate(BaseModel):
    account_type: str = "Organization"
    account: Optional[str] = None
    contact_name: Optional[str] = None
    contacts: Optional[List[dict]] = None
    industry: Optional[List[str]] = None
    industry_ids: Optional[List[int]] = None
    job_title: str
    date: Optional[str] = None
    job_url: Optional[str] = None
    salary_range: Optional[str] = None  # Legacy, kept for backwards compat
    salary_range_id: Optional[int] = None
    description: Optional[str] = None
    resume: Optional[str] = None
    status: str = "applied"
    source: str = "manual"
    project_ids: Optional[List[int]] = None


class JobUpdate(BaseModel):
    account: Optional[str] = None
    contacts: Optional[List[dict]] = None
    job_title: Optional[str] = None
    date: Optional[str] = None
    job_url: Optional[str] = None
    salary_range: Optional[str] = None  # Legacy, kept for backwards compat
    salary_range_id: Optional[int] = None
    description: Optional[str] = None
    resume: Optional[str] = None
    status: Optional[str] = None
    source: Optional[str] = None
    lead_status_id: Optional[str] = None
    project_ids: Optional[List[int]] = None

File: modules/agent/internal/schemas/job.go
package schemas

// JobCreate represents the request body for creating a job
type JobCreate struct {
	AccountType   string    `json:"account_type" validate:"omitempty,oneof=Organization Individual"`
	Account       *string   `json:"account"`
	ContactName   *string   `json:"contact_name"`
	Contacts      []Contact `json:"contacts"`
	Industry      []string  `json:"industry"`
	IndustryIDs   []int64   `json:"industry_ids"`
	JobTitle      string    `json:"job_title" validate:"required"`
	Date          *string   `json:"date"`
	JobURL        *string   `json:"job_url" validate:"omitempty,url"`
	DatePosted    *string   `json:"date_posted"`
	SalaryRange   *string   `json:"salary_range"`
	SalaryRangeID *int64    `json:"salary_range_id"`
	Description   *string   `json:"description"`
	Resume        *string   `json:"resume"`
	Status        string    `json:"status"`
	Source        string    `json:"source" validate:"omitempty,oneof=manual inbound referral event campaign agent"`
	ProjectIDs    []int64   `json:"project_ids"`
}

// JobUpdate represents the request body for updating a job
type JobUpdate struct {
	Account       *string   `json:"account"`
	Contacts      []Contact `json:"contacts"`
	JobTitle      *string   `json:"job_title"`
	Date          *string   `json:"date"`
	JobURL        *string   `json:"job_url" validate:"omitempty,url"`
	DatePosted    *string   `json:"date_posted"`
	SalaryRange   *string   `json:"salary_range"`
	SalaryRangeID *int64    `json:"salary_range_id"`
	Description   *string   `json:"description"`
	Resume        *string   `json:"resume"`
	Status        *string   `json:"status"`
	Source        *string   `json:"source"`
	LeadStatusID  *string   `json:"lead_status_id"`
	ProjectIDs    []int64   `json:"project_ids"`
}

// Contact represents a contact in job creation
type Contact struct {
	FirstName string  `json:"first_name"`
	LastName  string  `json:"last_name"`
	Email     *string `json:"email"`
	Phone     *string `json:"phone"`
	Title     *string `json:"title"`
}
